Assets/Scripts/UI/Patrolling/CommunicationZoneVertices.cs

```diff
@@ -0,0 +1,97 @@
+//
+// Contributors 2025: 
+// Casper Nyvang Sørensen,
+// Christian Ziegler Sejersen,
+// Jakob Meyer Olsen
+using System.Collections.Generic;
+using System.Linq;
+
+using Maes.Map;
+using Maes.Map.MapGen;
+using Maes.Robot;
+using Maes.Utilities;
+
+using UnityEngine;
+
+namespace Maes.UI.Patrolling
+{
+    public class CommunicationZoneVertices
+    {
+        public Dictionary<int, HashSet<int>> CommunicationZoneTiles { get; }
+        public HashSet<int> AllCommunicationZoneTiles { get; } = new();
+        private readonly SimulationMap<Tile> _simulationMap;
+        private readonly PatrollingMap _patrollingMap;
+        private readonly CommunicationManager _communicationManager;
+        private const int TrianglesPerCell = 8;
+
+        public CommunicationZoneVertices(SimulationMap<Tile> simulationMap, PatrollingMap patrollingMap, CommunicationManager communicationManager)
+        {
+            _simulationMap = simulationMap;
+            _patrollingMap = patrollingMap;
+            CommunicationZoneTiles = _patrollingMap.Vertices.ToDictionary(v => v.Id, _ => new HashSet<int>());
+            _communicationManager = communicationManager;
+        }
+
+        public void CreateCommunicationZoneTiles()
+        {
+            var positions = new List<Vector2Int>();
+            foreach (var vertex in _patrollingMap.Vertices)
+            {
+                positions.Add(vertex.Position);
+            }
+            var communicationZones = _communicationManager.CalculateCommunicationZone(positions, _simulationMap.WidthInTiles, _simulationMap.HeightInTiles);
+            var cellIndexToTriangleIndexes = CellIndexToTriangleIndexes(_simulationMap);
+            using var bitmap = MapUtilities.MapToBitMap(_simulationMap);
+
+            foreach (var vertex in _patrollingMap.Vertices)
+            {
+                var tiles = communicationZones[vertex.Position];
+                if (tiles == null)
+                {
+                    continue;
+                }
+                foreach (var tile in tiles)
+                {
+                    var index = tile.x + tile.y * bitmap.Width;
+                    CommunicationZoneTiles[vertex.Id].UnionWith(cellIndexToTriangleIndexes[index]);
+                }
+                AllCommunicationZoneTiles.UnionWith(CommunicationZoneTiles[vertex.Id]);
+            }
+        }
+
+        private static List<List<int>> CellIndexToTriangleIndexes(SimulationMap<Tile> simulationMap)
+        {
+            var cellIndexTriangleIndexes = new List<List<int>>();
+
+            var list = new List<int>();
+            foreach (var (index, _) in simulationMap)
+            {
+                list.Add(index);
+                if ((index + 1) % TrianglesPerCell == 0)
+                {
+                    cellIndexTriangleIndexes.Add(list);
+                    list = new List<int>();
+                }
+            }
+
+            return cellIndexTriangleIndexes;
+        }
```

Comment:
Make a utility class that contains this method, because this code is also used in LineOfSightVertices.cs -> reduce code duplication